pkg/client/client.go

```diff
@@ -120,6 +123,10 @@ func (r *CortexClient) doRequest(path, method string, payload []byte) (*http.Res
 		req.SetBasicAuth(r.id, r.key)
 	}
 
+	if r.authToken != "" {
+		req.Header.Add("Authorization", r.authToken)
+	}
+
```

Comment:
Should we error out if both token and basic auth are set?

Prometheus does the same https://github.com/prometheus/common/blob/main/config/http_config.go#L220-L242